tests/serverpod_test_server/test/encode_with_type_test.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var protocol = Protocol();
+
+  group('Given a', () {
+    test(
+        'non-null integer when encoding then output is the type name and value as a JSON string',
```

Comment:
Suggestion(NIT): Remove non-null since it is not nullable. I think it is enought with just `Given an integer...` here.